app/src/main/java/com/iic/mokojin/MainActivity.java

```diff
@@ -24,14 +30,23 @@ public boolean onCreateOptionsMenu(Menu menu) {
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        // Handle action bar item clicks here. The action bar will
-        // automatically handle clicks on the Home/Up button, so long
-        // as you specify a parent activity in AndroidManifest.xml.
         int id = item.getItemId();
 
-        //noinspection SimplifiableIfStatement
         if (id == R.id.action_settings) {
             return true;
+        } else if (id == R.id.action_goodnight){
+            final ProgressHudDialog dialog = new ProgressHudDialog(this, getString(R.string.good_night_progress));
+            dialog.show();
+            GoodNightOperation goodNightOperation = new GoodNightOperation();
+            goodNightOperation.run().continueWith(new Continuation<Void, Object>() {
+                @Override
+                public Object then(Task<Void> task) throws Exception {
+                    dialog.dismiss();
+                    CurrentSessionStore.get(MainActivity.this).refreshData();
+                    return null;
+                }
+            });
+            return true;
```

Comment:
Maybe this block needs to be extracted as a method `performGoodNightOperationWithProgressDialog`? For readability